### ISSUE TITLE
Add incoming socket attributes to http_request

### ISSUE THREAD

**Issue by chuckwolber** [microsoft/cpprestsdk#1380]
When a request comes into an http_listener object, an http_request object is generated that does not have access to any of the incoming socket information, such as the client's IP address. Having this information is helpful for generating logs.

**Comment by garethsb-sony** [comment]
Is `http_msg::remote_address` not what you want? https://github.com/microsoft/cpprestsdk/blob/f4124a9ee10042582c795578d69022368877282e/Release/include/cpprest/http_msg.h#L1027-L1031

**Comment by chuckwolber** [comment]
It is, but I think there may be more worth discussing here.

After a wave of self-imposed confusion, tracing through the source code cleared things up for me. But it also helped me to understand that the design philosophy of cpprestsdk was to hide the client request socket from the support handler, **and rightly so**. If you truly need that sort of access, you probably do not need the abstraction that cpprestsdk is providing.

The impetus behind my request was to start a discussion about what else can be provided to the support handler regarding the socket level details (IP being just one of them) of the incoming client request. 

In my experience, I rarely need more than the client IP address (which is almost certainly nat'd anyway), so this discussion is potentially moot. But there may be some benefit to considering other details like the incoming ephemeral port, etc.

**Comment by garethsb-sony** [comment]
> the design philosophy of cpprestsdk was to hide the client request socket from the support handler, **and rightly so.** If you truly need that sort of access, you probably do not need the abstraction that cpprestsdk is providing.

Yes, I also think that's the case, although I agree that having access to the underlying socket would be useful in several circumstances.
